src/corehost/cli/hostfxr.h

```diff
@@ -7,20 +7,35 @@
 #include <pal.h>
 
 // Forward declaration of required custom feature APIs
-using hostfxr_main_fn = int(*)(const int argc, const pal::char_t* argv[]);
-using hostfxr_main_startupinfo_fn = int(*)(
+using hostfxr_main_fn = int32_t(*)(const int argc, const pal::char_t* argv[]);
+using hostfxr_main_startupinfo_fn = int32_t(*)(
     const int argc,
     const pal::char_t* argv[],
     const pal::char_t* host_path,
     const pal::char_t* dotnet_root,
     const pal::char_t* app_path);
-using hostfxr_get_delegate_fn = int(*)(
+
+enum class hostfxr_delegate_type
+{
+    com_activation,
+    load_in_memory_assembly
+};
+
+using hostfxr_get_delegate_fn = int32_t(*)(
+    const pal::char_t* host_path,
+    const pal::char_t* dotnet_root,
+    const pal::char_t* app_path,
+    hostfxr_delegate_type type,
+    void** delegate);
+
+using hostfxr_get_delegate_fn_old = int(*)(
```

Comment:
Why is this one needed?

Reply:
I thought I removed that before committing...